### ISSUE TITLE
Tyler's recommended Github workflow steps.

### ISSUE THREAD

**Issue by jkl-semicolon** [more-travel-4u/more-travel-4u#91]
# 🦜 Tyler's Github Workflow steps 🦜 

Tyler came talked about his preferred workflow steps when it came to pushing and pulling code on Github, and how to handle merge conflicts. I think it may be a good idea to adopt his method and I will be doing that; but of course everyone should do what they feel comfortable with. The steps are detailed below:

_**Note: in the below example, let's assume we will work on imaginary ticket # 35, which is "adding comments to index.js"**_

1. Before starting a ticket, from branch main, pull the latest changes from main to your local repo and install any new packages. `git pull origin main` `npm install`
2. After choosing a ticket to work on, checkout to a new branch `git checkout -b '35-comment-index'`
3. Do work pertaining to your ticket on branch 35-comment-index.
4. After finishing your ticket, add and commit your changes to that branch. `git add .` `git commit -m 'added comments to index.js'`
5. Navigate back to your main branch locally. `git checkout main`
6. Pull the latest from main on Github, install any new packages. `git pull origin main` `npm install`
7. Navigate back to your working branch. `git checkout 35-comment-index`
8. Merge the latest from main locally onto your working branch. `git merge main`

>[!Important]
>At this point, there may be merge conflicts. Resolve the merge conflicts directly in your VS Code. Once any merge conflicts have been resolved, continue to step 9.

9. Push your branch to Github, first re-committing any changes that need to be re-committed. `git add .` `git commit -m 'resolved merge conflicts'` `git push origin 35-comment-index`
10.  On Github, submit a Pull Request for the changes made to branch 35-comment-index. Comment on what has been added/changed/deleted in your Pull Request description.

That's all the steps! To work on a new ticket, remember to navigate back to main, and then start over from Step 1. `git checkout main`
